regex replace ^$ pattern not searching around

**Describe the bug**
When doing a regex replace, the line beginnings and line endings are not replaced as multiline and global. Please see the capture.

**To Reproduce**
Please see capture

![regex_replace](https://user-images.githubusercontent.com/30809170/92753348-cb7ea380-f357-11ea-987c-cfd9725082a4.gif)

**Expected behavior**
It should replace all line endings and beginnings.

**Desktop (please complete the following information):**
 - OS: Version	10.0.19041 Build 19041
 - Version 1.4.2.0

I wanted to and I tried to compile it on my machine but for some reason, I'm getting this:

```
Notepads.Controls\Helpers\ThemeListener.cs(79,35,79,57): error CS0121:  The call is ambiguous.
```

between these two overloads:

```
'DispatcherQueueHelper.ExecuteOnUIThreadAsync(DispatcherQueue, Action, DispatcherQueuePriority)' 
'DispatcherQueueHelper.ExecuteOnUIThreadAsync(DispatcherQueue, Func<Task>, DispatcherQueuePriority)'
```

the call currently in use seems to be:

```
ExecuteOnUIThreadAsync(Action, CoreDispatcherPriority)
```

```
79: await DispatcherQueue.ExecuteOnUIThreadAsync(UpdateProperties, DispatcherQueuePriority.Normal);
```

In short, I didn't want to send a PR without being able to test it myself. This is probably something wrong with my setup as it is my first attempt with VS and C#.

--- a/src/Notepads/Controls/TextEditor/TextEditorCore.FindAndReplace.cs
+++ b/src/Notepads/Controls/TextEditor/TextEditorCore.FindAndReplace.cs
@@ -252,8 +252,9 @@
             try
             {
                 regexError = false;
+                content = content.Replace(RichEditBoxDefaultLineEnding, RegexDefaultLineEnding);
                 Regex regex = new Regex(searchContext.SearchText,
-                    RegexOptions.Compiled | (searchContext.MatchCase ? RegexOptions.None : RegexOptions.IgnoreCase));
+                    RegexOptions.Multiline | (searchContext.MatchCase ? RegexOptions.None : RegexOptions.IgnoreCase));
 
                 var match = regex.Match(content, Document.Selection.EndPosition);
 
@@ -287,8 +288,9 @@
             try
             {
                 regexError = false;
+                content = content.Replace(RichEditBoxDefaultLineEnding, RegexDefaultLineEnding);
                 Regex regex = new Regex(searchContext.SearchText,
-                    RegexOptions.RightToLeft | RegexOptions.Compiled |
+                    RegexOptions.RightToLeft | RegexOptions.Multiline |
                     (searchContext.MatchCase ? RegexOptions.None : RegexOptions.IgnoreCase));
 
                 var match = regex.Match(content, Document.Selection.StartPosition);
@@ -322,10 +324,11 @@
         {
             regexError = false;
             output = string.Empty;
+            content = content.Replace(RichEditBoxDefaultLineEnding, RegexDefaultLineEnding);
 
             try
             {
-                Regex regex = new Regex(searchContext.SearchText, RegexOptions.Compiled | (searchContext.MatchCase ? RegexOptions.None : RegexOptions.IgnoreCase));
+                Regex regex = new Regex(searchContext.SearchText, RegexOptions.Multiline | (searchContext.MatchCase ? RegexOptions.None : RegexOptions.IgnoreCase));
 
                 if (regex.IsMatch(content))
                 {
@@ -334,7 +337,7 @@
                         replaceText = ApplyTabAndLineEndingFix(replaceText);
                     }
 
-                    output = regex.Replace(content, replaceText);
+                    output = regex.Replace(content, replaceText).Replace(RegexDefaultLineEnding, RichEditBoxDefaultLineEnding);
                     return true;
                 }
             }

--- a/src/Notepads/Controls/TextEditor/TextEditorCore.cs
+++ b/src/Notepads/Controls/TextEditor/TextEditorCore.cs
@@ -35,6 +35,7 @@
         public event EventHandler<ScrollViewerViewChangingEventArgs> ScrollViewerViewChanging;
 
         private const char RichEditBoxDefaultLineEnding = '\r';
+        private const char RegexDefaultLineEnding = '\n';
 
         private bool _isDocumentLinesCachePendingUpdate = true;
         private string[] _documentLinesCache; // internal copy of the active document text in array format

